mlir/lib/Conversion/ArithToLLVM/ArithToLLVM.cpp

```diff
@@ -54,6 +54,23 @@ struct ConstrainedVectorConvertToLLVMPattern
   }
 };
 
+/// No-op bitcast.
+struct IdentityBitcastLowering final
+    : public OpConversionPattern<arith::BitcastOp> {
+  using OpConversionPattern::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(arith::BitcastOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const final {
+    Value src = adaptor.getIn();
+    if (src.getType() != getTypeConverter()->convertType(op.getType()))
```

Comment:
I find this a bit difficult to follow. Would it make sense to change this to: `if (!isa<MemRefType>(src.getType())`? And rename the pattern to `MemRefBitcastLowering`?

Reply:
Not sure if we need any memref-specific logic here, just handling same input/output converted types should be enough.